### ISSUE TITLE
Accessibility issue: Expanded change of state of filter button in igGrid does not announce

### ISSUE THREAD

**Issue by kousikgithubac** [IgniteUI/ignite-ui#2243]
## Description  
The expanded state change of filter button in igGrid needs to announce. This is in accordance with following guidelines:

**Rule details**
Expanded change of state not announced

**Rule ID**
man-sr-25

**WCAG success criteria**
4.1.2 - Name, Role, Value

**WCAG conformance level**
WCAG 2.0 Level A

**Disabilities affected**
Visual
Cognitive

* Ignite UI version: Any
* Browser: Any

## Steps to reproduce:
1. Go to the [sample](https://jsfiddle.net/georgianastasov/kn7ru52s/2/).
2. Use NVDA or another accessibility tool.
3. Navigate to the input field in the data grid.

## Result:
The state of the button "Filter", whether its content is collapsed or expanded, is not announced by the screen reader.

## Expected result:
The screen reader should announce when the state changes between expanded and collapsed.

## Attachments  
![Menu Expand State](https://github.com/IgniteUI/ignite-ui/assets/157089439/492fc121-17c2-4fe3-b38f-1f39ad36174b)

**Comment by ChronosSF** [comment]
@kousikgithubac  , thank you for the report! We are putting this for development for the first SR after the 2024.1 volume release. We expect to be able to deliver a fix by the end of August.

**Comment by ChronosSF** [comment]
Should be fixed by combining the fixes for #2241  and #2245
